source/class/qx/tool/cli/config/CompilerConfig.js

```diff
@@ -0,0 +1,111 @@
+/* ************************************************************************
+ *
+ *    qooxdoo-compiler - node.js based replacement for the Qooxdoo python
+ *    toolchain
+ *
+ *    https://github.com/qooxdoo/qooxdoo-compiler
+ *
+ *    Copyright:
+ *      2011-2019 Zenesis Limited, http://www.zenesis.com
+ *
+ *    License:
+ *      MIT: https://opensource.org/licenses/MIT
+ *
+ *      This software is provided under the same licensing terms as Qooxdoo,
+ *      please see the LICENSE file in the Qooxdoo project's top-level directory
+ *      for details.
+ *
+ *    Authors:
+ *      * John Spackman (john.spackman@zenesis.com, @johnspackman)
+ *
+ * *********************************************************************** */
+
+const path = require("path");
+const fs = qx.tool.utils.Promisify.fs;
+
+/**
+ * Loads and stores the configuration for the entire compiler
+ * 
+ */
+qx.Class.define("qx.tool.cli.config.CompilerConfig", {
+  extend: qx.tool.cli.config.AbstractConfig,
+  
+  construct: function(command) {
+    this.base(arguments);
+    this.__command = command;
+    this.__libraryConfigs = {};
+  },
+  
+  properties: {
+    configFilename: {
```

Comment:
Is this used anywhere? I think here, too, the config API should be used for consistency.

Reply:
It's used in the default implementation, and it says what the compiler thinks the .json file is should be called; this can be overridden by the command line args which is why it is passed in here.  The config API has been used as part of determining what this should be (in MConfig)